samza-core/src/main/scala/org/apache/samza/checkpoint/CheckpointTool.scala

```diff
@@ -173,13 +173,12 @@ class CheckpointTool(newOffsets: TaskNameToCheckpointMap, coordinatorStreamStore
     combinedConfigMap.putAll(userDefinedConfig)
     val combinedConfig: Config = new MapConfig(combinedConfigMap)
 
+    val taskConfig = new TaskConfig(combinedConfig)
     // Instantiate the checkpoint manager with coordinator stream configuration.
-    val checkpointManager: CheckpointManager = combinedConfig.getCheckpointManagerFactory() match {
-      case Some(className) =>
-        Util.getObj(className, classOf[CheckpointManagerFactory])
-          .getCheckpointManager(combinedConfig, new MetricsRegistryMap)
-      case _ =>
-        throw new SamzaException("Configuration: task.checkpoint.factory is not defined.")
+    val checkpointManager: CheckpointManager =
+      JavaOptionals.toRichOptional(taskConfig.getCheckpointManager(new MetricsRegistryMap)).toOption match {
```

Comment:
Would prefer to remove this RichOptional / JavaOptionals class / transformation from ScalaJavaUtil. I don't think it's adding any significant functionality, and is adding yet another "Option" class to keep track of.

For this PR, it's OK to just not use it. It's also OK to use it if you think this will be easy to refactor away later.

Reply:
Ah, yes, it can be removed here. I thought that Scala functions didn't work cleanly with Java classes, but it's not a problem here. I think there are still some places other where JavaOptionals might be useful, since sometimes it is nicer to use Scala syntax when in a Scala class.

Reply:
Oops, I take that last comment back. I guess Scala functions aren't compatible with Java functions.
For example, `val supplier: Supplier[String] = () => "hello world"` doesn't actually compile (IDE doesn't show the red squiggle, but build actually will have an error).
So having an `Option` will allow for more natural syntax usage in Scala.